nixos/attestation-server/module.nix

```diff
@@ -64,6 +94,22 @@ in
 
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/AttestationServer";
+        ExecStartPre = let
+          values = concatStringsSep ", " [
+            "('emailUsername', '${cfg.emailUsername}')"
+            "('emailPassword', TRIM(readfile('%S/attestation/emailPassword'), char(10)))"
+            "('emailHost', '${cfg.emailHost}')"
+            "('emailPort', '${toString cfg.emailPort}')"
+          ];
+        in optionals (cfg.emailPasswordFile != null) [
+          "+${pkgs.coreutils}/bin/install -m 0600 -o %N -g %N ${cfg.emailPasswordFile} %S/attestation/emailPassword"
```

Comment:
Note the leading `+`.  This command has to run as `root` outside the sandbox because `emailPasswordFile` could be about anywhere in the file system.  As user and group I use `%N` which resolves to the unit name.  Unfortunately it seems that there is no way to access the user and group that the service runs as (`%u` and `%g` resolve to user and group of the service **manager** which is `root` and `root`).

Reply:
Could you add as a comment in the source?

Reply:
Done.